app/hotels/src/allHotels/AllHotels.js

```diff
@@ -17,53 +15,69 @@ type Props = {|
   onFilterChange: (filter: SearchParametersType) => void,
 |};
 
-export default class AllHotelsSearch extends React.Component<Props> {
+export default class AllHotels extends React.Component<Props> {
+  /**
+   * Submit form with the initial date interval parameters.
+   */
+  componentDidMount = () => {
+    this.props.onFilterChange({
+      checkin: moment()
+        .add(1, 'week')
+        .startOf('isoWeek')
+        .format('YYYY-MM-DD'),
+      checkout: moment()
+        .add(1, 'week')
+        .endOf('isoWeek')
+        .format('YYYY-MM-DD'),
+    });
+  };
+
   handleSearchChange = (search: SearchParametersType) => {
     this.props.onFilterChange(search);
   };
 
-  isReadyToSearch = (): boolean => {
-    const { search: s } = this.props;
+  renderInnerComponent = rendererProps => {
+    const {
+      onFilterChange,
+      openSingleHotel,
+      search,
+      location,
+      onLocationChange,
+    } = this.props;
+    const data = idx(rendererProps, _ => _.props.city) || null;
+
     return (
-      s.latitude !== null &&
-      s.longitude !== null &&
-      s.checkin !== null &&
-      s.checkout !== null
+      <AllHotelsSearch
+        data={data}
+        search={search}
+        location={location}
+        onFilterChange={onFilterChange}
+        onLocationChange={onLocationChange}
+        openSingleHotel={openSingleHotel}
+      />
     );
   };
 
-  renderInnerComponent = (propsFromRenderer: AllHotelsSearchQueryResponse) => (
-    <AllHotelsSearchList
-      data={propsFromRenderer.allHotels}
-      openSingleHotel={this.props.openSingleHotel}
-    />
-  );
-
   render = () => {
-    const { search } = this.props;
+    const { location } = this.props;
+
     return (
-      <Layout>
-        <SearchForm onChange={this.handleSearchChange} />
-        <FilterStripe />
-        {this.isReadyToSearch() && (
-          <PublicApiRenderer
-            query={graphql`
-              query AllHotelsSearchQuery($search: HotelsSearchInput!) {
-                allHotels: allAvailableHotels(search: $search) {
-                  ...AllHotelsSearchList
-                }
-              }
-            `}
-            variables={{
-              search: search,
-            }}
-            render={this.renderInnerComponent}
-            cacheConfig={{
-              force: true,
-            }}
-          />
-        )}
-      </Layout>
+      <SimpleQueryRenderer
+        query={graphql`
+          query AllHotelsQuery($prefix: String!) {
+            city: hotelCities(prefix: $prefix, first: 1) {
+              ...AllHotelsSearch
+            }
+          }
+        `}
+        render={this.renderInnerComponent}
+        variables={{
+          prefix: location,
+        }}
+        cacheConfig={{
+          force: true,
```

Comment:
We can probably save the result of this query into the offline cache. Does it make sense?